docs/specs/workflow.md

```diff
@@ -318,3 +318,99 @@ And then the frontend will receive a response to telling it to redirect.
+
+### Intent for login id migration (Experimental)
+
+The Intent for login id migration allows the developer to create a workflow for migration.
+
+[![](https://mermaid.ink/img/pako:eNqdUl1rwkAQ_CvLPdv2PQ9Cq4JS_EArfWhK2SYbc2ju0ru9ioj_vZvGaCVgSwMHuZ2ZnZu926vEpqQi5ekjkEmor3HlsIgNyFeiY53oEg3DfeB8RejaSC94tsVy1EaG1q4BPfTJWLiDZ3rPpVLzGtVNt9u0jqDnCJlAGybDYy0nkd3WunW2sdta15BF17SIYE4cnDkxYdTvSBPPKIlkc8Vx5mxClALnZyPYas7hafo4mLQ8q0iiQu9rArCtY35qhHO6qnRi96wcxYWE4eVVp5JM8-72mG5RUgJoUoFQXCowQbbuAs-cLWA2nE4Gb5Pl-GEwvzA5hzmO4X8urah_GVJV8SL2v1_Otfv4SfahoAunil0t1VEFuQJ1Ku91_11VwisoVpH8ppRh2HCsYnMQquS0i51JVCSjp44KZSpBj89bRRluPB2-AAamCc8?type=png)](https://mermaid.live/edit#pako:eNqdUl1rwkAQ_CvLPdv2PQ9Cq4JS_EArfWhK2SYbc2ju0ru9ioj_vZvGaCVgSwMHuZ2ZnZu926vEpqQi5ekjkEmor3HlsIgNyFeiY53oEg3DfeB8RejaSC94tsVy1EaG1q4BPfTJWLiDZ3rPpVLzGtVNt9u0jqDnCJlAGybDYy0nkd3WunW2sdta15BF17SIYE4cnDkxYdTvSBPPKIlkc8Vx5mxClALnZyPYas7hafo4mLQ8q0iiQu9rArCtY35qhHO6qnRi96wcxYWE4eVVp5JM8-72mG5RUgJoUoFQXCowQbbuAs-cLWA2nE4Gb5Pl-GEwvzA5hzmO4X8urah_GVJV8SL2v1_Otfv4SfahoAunil0t1VEFuQJ1Ku91_11VwisoVpH8ppRh2HCsYnMQquS0i51JVCSjp44KZSpBj89bRRluPB2-AAamCc8)
+
+#### Configuration in `authgear.yaml`
+
+```yaml
+migration_workflow:
+  url: authgeardeno:///deno/migration.ts
+  timeout: 10
+```
+
+The developer can pass the request body to the workflow. Authgear will send the request body to the hook, and the hook should return the identities' and authenticators' MigrateSpec.
+
+In the below examples, the developer put the migration token (generated by the developer) into the request body.
+
+#### Create the migration intent
+
+```json
+{
+  "intent": {
+    "kind": "IntentMigrateLoginID",
+    "data": {
+      "request_body": "TOKEN"
+    }
+  }
+}
+```
+
+#### Hook response
+
+```json
+{
+  "identities": [
+    {
+      "type": "login_id",
+      "login_id": {
+        "key": "email",
+        "type": "email",
+        "value": "test@example.com"
+      }
+    }
+  ],
+  "authenticators": [
+    {
+      "type": "oobotp",
+      "is_default": true,
+      "kind": "primary",
+      "oobotp": {
+        "email": "test@example.com"
+      }
+    }
+  ]
+}
+```
+
+The developer has the responsibility to ensure the identities and authenticators combination is valid.
+Otherwise, the user won't be able to login with the identity. e.g. login id identity should have its primary authenticator.
+
+Only login id identity and OOB OTP authenticator are supported in this stage.
+
+#### The identity spec
+
+```go
+
+type Spec struct {
+	Type model.IdentityType `json:"type"`
+
+	LoginID *LoginIDSpec `json:"login_id,omitempty"`
+}
+
+type LoginIDSpec struct {
+	Key   string               `json:"key"`
+	Type  model.LoginIDKeyType `json:"type"`
+	Value string               `json:"value"`
+}
+```
+
+### The authenticator spec
+```go
+type Spec struct {
+	Type      model.AuthenticatorType `json:"type,omitempty"`
+	IsDefault bool                    `json:"is_default,omitempty"`
```

Comment:
I think the developer has no idea when do they set this to false. I guess we should omit it.